### DISCUSSION TITLE
Webservice endpoints plurality?

### DISCUSSION THREAD

**Discussion by pjdevries** [joomla/joomla-cms#34080] [Q&A]
Is there a specific reason why endpoint `v1/content/article`, to request a list of articles, is singular and `v1/content/categories`, to request a list of categories, is plural? Even more confusing, `v1/fields/content/articles` is also plural.

**Comment by pjdevries** [comment]
To be more clear and avoid confusion: I am not debating whether it should be singular or plural. I am just wondering why we have this inconsistency.

**Comment by Fedik** [comment]
I do not know the reason,
but I suspect it is coincidence

**Comment by alikon** [comment]
so `v1/content/article` should be `v1/content/articles`  for consistency right ?

also 
`v1/content/article/contenthistory/` should be `v1/content/articles/contenthistory/`
`v1/content/article/contenthistory/keep` should be `v1/content/articles/contenthistory/keep`

**Comment by alikon** [comment]
please test https://github.com/joomla/joomla-cms/pull/34093

**Comment by joomdonation** [comment]
+1 for using plural.

**Comment by wilsonge** [comment]
The original spec we did in 2017 GSOC was singular (ref: https://joomla-projects.github.io/gsoc19_webservices/?specification/chapters/urls-and-routing.md).

Generally speaking though as long as it's consistent I don't mind hugely. It's worth noting though that the type of the object returned in the JSON API payload MUST be singular - so if we want to do plural URLs it's worth noting that the type in the body of the resource will always still be the singular version.

**Reply by joomdonation** [reply]
For some systems I have worked on like Mailchimp, Bubspot... (integration with their REST API only for my extensions) , they always use plural (even WordPress use plural, too). So maybe it is a standard or  best practice :).
